File: docs/guides/dart/lib/useConditionlessRule.dart
import 'package:algolia_client_search/algolia_client_search.dart';

void useConditionlessRule() async {
  final client =
      SearchClient(appId: 'ALGOLIA_APPLICATION_ID', apiKey: 'ALGOLIA_API_KEY');

  final objectID = "a-rule-id";

  final rule = Rule(
      objectID: objectID,
      consequence: Consequence(
          // Set relevant consequence
          ),
      // Set validity (optional)
      validity: [TimeRange(from: 1688774400, until: 1738972800)]);

  await client.saveRule(
    indexName: "<YOUR_INDEX_NAME>",
    objectID: objectID,
    rule: rule,
  );
}

File: docs/guides/csharp/src/useConditionlessRule.cs
namespace Algolia;

using System;
using System.Collections.Generic;
using System.Net.Http;
using System.Text.Json;
using Algolia.Search.Clients;
using Algolia.Search.Http;
using Algolia.Search.Models.Search;

class UseConditionlessRule
{
  async Task Main(string[] args)
  {
    var client = new SearchClient(new SearchConfig("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY"));

    var objectID = "a-rule-id";

    var rule = new Rule
    {
      ObjectID = objectID,
      Consequence =
        new Consequence( /* Set relevant consequence */
        ),
      // Set validity (optional)
      Validity = [new TimeRange { From = 1688774400L, Until = 1738972800L }],
    };

    await client.SaveRuleAsync("<YOUR_INDEX_NAME>", objectID, rule);
  }
}
